modules/internal/ChapelArray.chpl

```diff
@@ -4334,9 +4334,21 @@ module ChapelArray {
   //
   // Swap operator for arrays
   //
-  inline proc <=>(x: [], y: []) {
-    forall (a,b) in zip(x, y) do
-      a <=> b;
+  inline proc <=>(x: [?xD], y: [?yD]) {
+    var hasSwapped: bool = false;
+    // Check if array can use optimized pointer swap
+    if (!chpl__isArrayView(x._value)
+        && x._value.type == y._value.type
+        && xD == yD) {
+      if(Reflection.canResolveMethod(x._value, "doiSwap", y._value)) {
```

Comment:
Can you reorder these checks so that static checks are first and the dynamic check (domain equality) is inside of them? I can't imagine a case where it'll generate different code, but it feels more principled that way.

Reply:
I'd also put ampersands at the end of the lines rather than beginning, but it may be just me